compiler/quilt/tools/compat.py

```diff
@@ -0,0 +1,23 @@
+"""
+Python 2 / 3 compatibility
+
+In cases where the `six` lib covers an item, use `six`.
+
+For others, import the module here under the python3 name (if there are naming differences).
+
+Unfortunately, this *doesn't* allow for `from .tools.compat.foo import bar`, so if needed,
+import frequently-used objects here for convenience.
+"""
+# unused imports -- pylint: disable=W0611
+
+import sys as _sys
+
+# XXX: This originally had a few modules, but I've managed to remove the others.
+#      I'm tentatively still keeping compat.py, otherwise the try: except: block ends up in
+#      each module's import stanzas, instead of 'from compat import pathlib'.
+
+try:
+    import pathlib2 as pathlib
+except ImportError:
+    assert _sys.version_info >= (3, 6)   # use `pip install --upgrade quilt` if needed
```

Comment:
I wouldn't bother with the assert - just try the two imports

Reply:
The assert is there because an outdated version of pathlib might be found and successfully imported, causing problems in an un-obvious way (far from import time).  Anyways, I reworded it so that the assert wasn't needed.